feat: support .NET

olljanat opened the pull request:
Targets to support .NET "publish" and "test" commands on same way like Golang build and test works.

~Marked as draft because unit test(s) to verify "test" command is not yet implemented.~

gerhard commented:
Thank you @olljanat for your first contribution! We just kicked off the workflow for you 👟

Would love to get your 👀 & 🙇‍♂️ on this one @cdhunt & @nabsul.

Should this be added to the experimental namespace - `universe.dagger.io/x/<MAINTAINER_EMAIL>/dotnet` - @shykes, same as #1991? If yes, then we should add this to https://github.com/dagger/.github/blob/main/CONTRIBUTING.md

cdhunt commented on pkg/universe.dagger.io/dotnet/build.cue:
```diff
@@ -0,0 +1,31 @@
+)
+
+// Build a dotnet binary
+#Build: {
```
Naming the action `#Build` when it calls `publish` could be confusing. I know you're following the existing Dagger convention, but my suggestion is to mimic the dotnet SDK command names.

olljanat commented:
Moved to experimental namespace, renamed `#Build` command to ´#Publish`, fixed cuelint issues and added support for `#Test` command. This should be ready for merge unless you find other issues from it.

olljanat commented on pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue:
```diff
@@ -0,0 +1,31 @@
+	// Target package to publish
+	package: *"." | string
+
+	env: [string]: string
```
Probably left over as I start from Go package. Will drop it as `dotnet publish` does not actually care about environment variables (afaik).

cdhunt replied:
`dotnet publish` doesn't, but MSBuild can/does and dotnet might call out to msbuild depending on the project configuration.

olljanat replied:
Valid point. Now environment variables should be available inside of container too.

aluzzardi reviewed (approved):
Thank for you for the contribution, @olljanat!

Looks good to merge after CI passes.

TomChv commented on pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue:
```diff
@@ -0,0 +1,34 @@
+
+	container: #Container & {
+		"source": source
+		"env": {
```
`"env": env` is enough ;)

TomChv reviewed (approved):
LGTM too, I left a style comment but it's mergeable after CI passes
